odata-url-conventions/5 Query Options.md

```diff
@@ -125,6 +128,19 @@ and only to itself.
 
 The special value `NaN` is not equal to anything, even to itself.
 
+::: example
+Example ##ex_implicitcast: Comparing values of different types
+```
+'4.0' eq 4
+```
+evalutes to false, because rule 2 for the [`cast`](#cast) function is used to cast
+both operands to strings, and
+```
+'4.0' eq '4'
+```
+is false.
+:::
+
```

Comment:
This is very peculiar. But the `cast` functions allows casting in both directions, so we need a precedence rule here. If casting rule 3 was applied, `4.0 eq 4` would be true.